staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr_test.go

```diff
@@ -324,3 +330,239 @@ func TestParse(t *testing.T) {
+				}
+				const want = "cbor: cannot unmarshal map into Go value of type int32"
+				if got := err.Error(); got != want {
+					t.Fatalf("want error message %q, got %q", want, got)
+				}
+
+			},
+		},
+		{
+			in:   []byte{0x40},
+			want: FromString(""),
+		},
+		{
+			in:   []byte{0x43, 'a', 'b', 'c'},
+			want: FromString("abc"),
+		},
+		{
+			in:   []byte{0x00},
+			want: FromInt32(0), // min positive integer representable in one byte
+		},
+		{
+			in:   []byte{0x17},
+			want: FromInt32(23), // max positive integer representable in one byte
+		},
+		{
+			in:   []byte{0x18, 0x18},
+			want: FromInt32(24), // min positive integer representable in two bytes
+		},
+		{
+			in:   []byte{0x18, 0xff},
+			want: FromInt32(math.MaxUint8), // max positive integer representable in two bytes
+		},
+		{
+			in:   []byte{0x19, 0x01, 0x00},
+			want: FromInt32(math.MaxUint8 + 1), // min positive integer representable in three bytes
+		},
+		{
+			in:   []byte{0x19, 0xff, 0xff},
+			want: FromInt32(math.MaxUint16), // max positive integer representable in three bytes
+		},
+		{
+			in:   []byte{0x1a, 0x00, 0x01, 0x00, 0x00},
+			want: FromInt32(math.MaxUint16 + 1), // min positive integer representable in five bytes
+		},
+		{
+			in:   []byte{0x1a, 0x7f, 0xff, 0xff, 0xff},
+			want: FromInt32(math.MaxInt32), // max positive integer representable by Go int32
+		},
+		{
+			in:   []byte{0x20},
+			want: FromInt32(-1), // max negative integer representable in one byte
+		},
+		{
+			in:   []byte{0x37},
+			want: FromInt32(-24), // min negative integer representable in one byte
+		},
+		{
+			in:   []byte{0x38, 0x18},
+			want: FromInt32(-1 - 24), // max negative integer representable in two bytes
+		},
+		{
+			in:   []byte{0x38, 0xff},
+			want: FromInt32(-1 - math.MaxUint8), // min negative integer representable in two bytes
+		},
+		{
+			in:   []byte{0x39, 0x01, 0x00},
+			want: FromInt32(-2 - math.MaxUint8), // max negative integer representable in three bytes
+		},
+		{
+			in:   []byte{0x39, 0xff, 0xff},
+			want: FromInt32(-1 - math.MaxUint16), // min negative integer representable in three bytes
+		},
+		{
+			in:   []byte{0x3a, 0x00, 0x01, 0x00, 0x00},
+			want: FromInt32(-2 - math.MaxUint16), // max negative integer representable in five bytes
+		},
+		{
+			in:   []byte{0x3a, 0x7f, 0xff, 0xff, 0xff},
+			want: FromInt32(math.MinInt32), // min negative integer representable by Go int32
+		},
+	} {
+		t.Run(fmt.Sprintf("{Type:%d,IntVal:%d,StrVal:%q}", tc.want.Type, tc.want.IntVal, tc.want.StrVal), func(t *testing.T) {
+			var got IntOrString
+			err := got.UnmarshalCBOR(tc.in)
+			if tc.assertOnError != nil {
+				tc.assertOnError(t, err)
+			} else if err != nil {
+				t.Fatalf("unexpected error: %v", err)
+			}
+			if diff := cmp.Diff(got, tc.want); diff != "" {
+				t.Errorf("unexpected difference between expected and actual output:\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestIntOrStringRoundtripCBOR(t *testing.T) {
```

Comment:
somewhat surprised this one works.  I expected to see a custom fuzzer that decides to set the int part or the string part instead of potentially using both.  Does this work reliably?

Reply:
It implements fuzz.Interface in `intstr_fuzz.go` (similar story for Time and MicroTime).

Reply:
ah

Reply:
Also, I took a look at the inputs it was generating to make sure these methods were actually being used:

```
=== RUN   TestIntOrStringRoundtripCBOR
    intstr_test.go:551: {0 1128564365 }
    intstr_test.go:551: {0 200166354 }
    intstr_test.go:551: {0 -276099788 }
    intstr_test.go:551: {0 -1249228784 }
    intstr_test.go:551: {0 1742569458 }
    intstr_test.go:551: {0 -1375958113 }
    intstr_test.go:551: {1 0 (jʢǔ腔誖4Þ[{謴ǡ嶆Jɏ嗑}
    intstr_test.go:551: {0 1906691861 }
    intstr_test.go:551: {1 0 TĆ帎ȼCǲ屁ɻEwû与塩衰噺瀷Ȋ椮}
    intstr_test.go:551: {0 1459274389 }
    intstr_test.go:551: {1 0 }
    intstr_test.go:551: {0 -467127729 }
    intstr_test.go:551: {0 -705730334 }
    intstr_test.go:551: {1 0 溿ŷʈ沙t:励¬Ⱦ}
    intstr_test.go:551: {1 0 鷩}
    intstr_test.go:551: {0 -126567898 }
    intstr_test.go:551: {0 953288002 }
    intstr_test.go:551: {0 -1326681651 }
    intstr_test.go:551: {1 0 謳嬅Ɵ<炋靼瓽M蜢ɣȷ}
    intstr_test.go:551: {1 0 *堗ƛ強ɵpŵŴȭ-ģ2}
    intstr_test.go:551: {0 -1009384221 }
    intstr_test.go:551: {1 0 ɖ昮ǀǝ&ǻ}ń窚X`'整}
    intstr_test.go:551: {0 1993804638 }
    intstr_test.go:551: {0 2066970653 }
    intstr_test.go:551: {1 0 筄D垞}
    intstr_test.go:551: {0 -1084349919 }
    intstr_test.go:551: {0 1653629984 }
    intstr_test.go:551: {0 784751765 }
    intstr_test.go:551: {0 620224278 }
    intstr_test.go:551: {1 0 uÑē靎ĽIɭÓ=謋Ƀ喠靬?Ǜ}
    intstr_test.go:551: {0 577636246 }
    intstr_test.go:551: {0 -650130851 }
    intstr_test.go:551: {1 0 @Ś4v}
```